### ISSUE TITLE
Large mismatch between bpp loss and bitstream bpp for low initial warmup learning rates

### ISSUE THREAD

**Issue by YodaEmbedding** [InterDigitalInc/CompressAI#180]
The default learning rate is 1e-4, which seems stable for most compression models.

When warming up with low initial learning rates (e.g. a ramp from 1e-6 to 1e-4), the training bpp_loss behaves reasonably. However, the actual encoded bitstream length goes to 10x the expected bpp amount. Perhaps the model is generating outputs or distributions that are poorly handled by the entropy coder.

Related: I have seen NaN occur for learning rates >1e-4. This is to be expected if the learning rate is sufficiently large. However, something as small as 2e-4 is capable of NaNing certain models. I briefly tried fiddling around with gradient clipping when using higher learning rates, but that did not seem to have an effect.
